src/OpenRiaServices.Server.UnitTesting/Framework/DomainServiceTestHost.cs

```diff
@@ -992,13 +992,14 @@ private OperationContext CreateOperationContext(DomainOperationType operationTyp
         {
             // TODO: consider whether this implementation should call dispose
             //  Maybe OperationContext would implement IDisposable...
+            
+            IPrincipal user = (IPrincipal)ServiceProvider.GetService(typeof(IPrincipal));
             DomainServiceContext domainServiceContext =
-                new DomainServiceContext(this.ServiceProvider, operationType);
+                new DomainServiceContext(this.ServiceProvider, user, operationType);
```

Comment:
Use none-obsolete constructor

Reply:
What do you think about making user a public property and use that?
The old behaviour where should be preserved (if ctor with only serviceprovider is used then try resolve the user in the constructor).

Reply:
Future work would be to allow the factory to be null and in those cases create the domainservice via the servicefactory (but would need to handle scopes)

Reply:
Updated